### PR TITLE
fix 'tuple index out of range' error in the fit.est_errors method

### PR DESCRIPTION
**Pull request by dtnguyen2**
# Release Note

A fix for issue #342.

Sherpa's ```Confidence``` class is confused if a parameter is frozen then thawed after a fit resulting with an exception of the form: ```IndexError: tuple index out of range```.  The error does not show up if conf is called from the ui level

### CODE REVIEW

File: sherpa/fit.py [hunk 1]
```diff
@@ -1052,6 +1045,16 @@ def __init__(self, data, model, stat=None, method=None, estmethod=None,
                                 itermethod_opts)
         NoNewAttributesAfterInit.__init__(self)
 
+    def calc_thaw_indices(self):
+        iter = 0
```
Comment (DougBurke) [inline sherpa/fit.py]: This routine can be simplified to

    self.thaw_indices = tuple([i for i, par in enumerate(self.model.pars) if not par.frozen])

(not tested, I should say), which I find a bit easier to follow.

Comment (DougBurke) [reply sherpa/fit.py]: Oh, I see I'd essentially given this as a workaround in https://github.com/sherpa/sherpa/issues/342#issuecomment-284405268

(but dropping the intermediate array in the comprehension - i.e.

    ... = tuple(i for i, ...)

**Comment by dtnguyen2** [comment]
made the change as discussed on Thu

**Comment by DougBurke** [comment]
I think the code is fine (for fun I ran a test run where the `calc_thaw_indices` method raised a NotImplementedError and a lot of tests failed, so it's definitely well tested), so the only things left I think are

 * changelog
 * I think we need to update the copyright date in `sherpa/fit.py` to include 2019

**Comment by dtnguyen2** [comment]
Doug must have added 2019 already

**Comment by DougBurke** [comment]
I think there's something a little odd in the history of this PR because whenever I switch to it it requires me to update the sherpa-test-data submodule, but there's no change to the test directory here. I tried rebasing it onto the master branch and there's a conflict with 'fit.py' (related to 85239eb I guess) but it also complains about the sherpa-test-data update - e.g. 

```
% diff --git a/sherpa-test-data b/sherpa-test-data
index 4000e17..3f92400 160000
--- a/sherpa-test-data
+++ b/sherpa-test-data
@@ -1 +1 @@
-Subproject commit 4000e170bf2e51591da6305a671af22ea4ecca26
+Subproject commit 3f924001d848187d1d21b3f8267d99e8b02c1c3c
```

(although I don't know which is from this PR and which is from the master branch in this diff output)

**Comment by DougBurke** [comment]
Note to self: We also need to update the sphinx documentation - `docs/fit/index.rst` and `docs/_examples/fit/index.py` - since they use this workaround.

**Comment by DougBurke** [comment]
Perhaps bad form to  add to the PR after approving it, but I've updated the sphinx documentation to remove the workaround for #342.

**Pull request merged**